src/server/wesnothd/player.cpp

```diff
@@ -34,6 +34,7 @@ wesnothd::player::player(const std::string& n, simple_wml::node& cfg,
 	cfg_.set_attr_dup("name", n.c_str());
 	cfg_.set_attr("registered", registered ? "yes" : "no");
 	cfg_.set_attr("moderator", moderator ? "yes" : "no");
+	cfg.set_attr_int("forum_id", id);
```

Comment:
Does this work correctly? You're setting on the function parameter and not the member.

Reply:
It does - I tested it by printing out the data sent to the client and the Profile button now in the MP lobby correctly creates the link using the forum ID.  It's a node reference to data owned by `games_and_users_list_`, so both refer to the same thing, but still definitely a typo.